homeassistant/components/subaru/sensor.py

```diff
@@ -263,16 +262,22 @@ def available(self):
         last_update_success = super().available
         if last_update_success and self.vin not in self.coordinator.data:
             return False
+        if self.state is None:
+            return False
         return last_update_success
 
     def get_current_value(self):
         """Get raw value from the coordinator."""
-        value = self.coordinator.data[self.vin][VEHICLE_STATUS].get(self.data_field)
-        if value in sc.BAD_SENSOR_VALUES:
-            value = None
-        if isinstance(value, str):
-            if "." in value:
-                value = float(value)
-            else:
-                value = int(value)
-        return value
+        if isinstance(self.coordinator.data, dict):
+            value = self.coordinator.data.get(self.vin)[VEHICLE_STATUS].get(
+                self.data_field
+            )
+            if value in sc.BAD_SENSOR_VALUES:
+                value = None
+            if isinstance(value, str):
```

Comment:
There are multiple unrelated changes in this PR.  Please pull them out into separate PRs

Reply:
Changes unrelated to locks have been removed.